jsapp/js/components/modalForms/dataAttachmentColumnsForm.es6

```diff
@@ -0,0 +1,226 @@
+      ),
+    );
+    this.setModalTitle();
+  }
+
+  componentWillUnmount() {
+    this.unlisteners.forEach((clb) => {clb();});
+  }
+
+  setModalTitle() {
+    this.props.onSetModalTitle(
+      t('Import data from ##PARENT_NAME##')
+        .replace('##PARENT_NAME##', this.props.parent.name)
+    );
+  }
+
+  onAttachToParentCompleted() {
+    this.props.onModalClose();
+  }
+  onAttachToParentFailed() {
+    this.setState({isLoading: false});
+  }
+  onBulkSelect() {
+    let newList = this.state.columnsToDisplay.map((item) => {
+      return {label: item.label, checked: true}
+    });
+    this.setState({columnsToDisplay: newList})
+  }
+  onBulkDeselect() {
+    let newList = this.state.columnsToDisplay.map((item) => {
+      return {label: item.label, checked: false}
+    });
+    this.setState({columnsToDisplay: newList})
+  }
+  onLoadAssetContentCompleted(response) {
+    if (
+      response.data_sharing?.fields.length > 0
+    ) {
+      this.setState({
+        isVirgin: false,
+        columnsToDisplay: this.props.generateColumnFilters(
+          this.props.fields,
+          response.data_sharing.fields,
+        ),
+      });
+    } else {
+      // empty `fields` implies all parent questions are exposed
+      this.setState({
+        isVirgin: false,
+        columnsToDisplay: this.props.generateColumnFilters(
+          this.props.fields,
+          response.content.survey,
+        ),
+      });
+    }
+  }
+  onPatchParentCompleted(response) {
+    this.setState({
+      isLoading: false,
+      columnsToDisplay: this.props.generateColumnFilters(
+        this.props.fields,
+        response.fields,
+      ),
+    });
+    this.props.onModalClose();
+  }
+  onPatchParentFailed() {
+    this.setState({isLoading: false});
+  }
+
+  onColumnSelected(newList) {
+    this.setState({columnsToDisplay: newList});
+  }
+
+  onSubmit(evt) {
+    evt.preventDefault();
+    this.setState({isLoading: true});
+    this.props.triggerParentLoading();
+
+    let fields = [];
+    var data = '';
+
+    this.state.columnsToDisplay.map((item) => {
+      if (item.checked) {
+        fields.push(item.label);
+      }
+    });
+
+    if (this.props.attachmentUrl) {
+      data = JSON.stringify({
+        fields: fields,
+        filename: this.props.filename,
+      });
+      actions.dataShare.patchParent(this.props.attachmentUrl, data);
+    } else {
+      data = JSON.stringify({
+        parent: this.props.parent.url,
+        fields: fields,
+        filename: this.props.filename,
+      });
+      actions.dataShare.attachToParent(this.props.asset.uid, data);
+    }
+  }
+
+  render() {
+    return (
+      <bem.FormModal__form m='data-attachment-columns'>
+        <div className='header'>
+          <span className='modal-description'>
+            {t('You are about to import ##PARENT_NAME##. Select or deselect in the list below to narrow down the number of questions to import.').replace('##PARENT_NAME##', this.props.parent.name)}
```

Comment:
I'm not sure if you guys try and stick to a text-width of 80, but maybe format this to something like the following (not sure if this works the same as in Python)?

```javascript
t(
  'You are about to import ##PARENT_NAME##. Select or deselect in',
  'the list below to narrow down the number of questions to import.'
).replace('##PARENT_NAME##', this.props.parent.name)
```

or

```javascript
t(
  `You are about to import ${this.props.parent.name}. Select or deselect`,
  'in the list below to narrow down the number of questions to import.'
)
```

Reply:
I try to stick to 80 and I wanted to do it like the way you're describing, but I'm not entirely sure. We want to set up `prettier.io` integration: https://github.com/kobotoolbox/kpi/issues/2983 and I want to follow their [rationale](https://prettier.io/docs/en/rationale.html#print-width) to prepare for this. In it they say 80 is a "guideline" and their formatter doesn't shorten long strings. But... clearly it's [a point of contention](https://github.com/prettier/prettier/issues/4123) so I'm not sure :sweat_smile: It's something I want to ask @magicznyleszek about though!

Reply:
I think we should leave them long. There is no great way to fit them in 80 characters - splitting it with `+` will break getting strings out from `t(` or make it complex to handle.